### PR TITLE
[LowerSwitch] Implement verifyAnalysis

### PR DESCRIPTION
**Pull request by jayfoad**
This pass is used like an analysis so it's useful to be able to verify
passes that claim to have preserved it.

### CODE REVIEW

File: llvm/lib/Transforms/Utils/LowerSwitch.cpp [hunk 1]
```diff
@@ -569,8 +569,17 @@ class LowerSwitchLegacyPass : public FunctionPass {
...
   }
 
+  // Remember the current function. Only used for verification.
+  Function *F;
+
   bool runOnFunction(Function &F) override;
 
+  void verifyAnalysis() const override {
+    for (auto &BB : *F)
+      assert(!isa<SwitchInst>(BB.getTerminator()) &&
```
Comment (arsenm) [inline llvm/lib/Transforms/Utils/LowerSwitch.cpp:579]: Is verifyAnalysis supposed to work in a release build? I think you're in danger of the unused member variable as is

Comment (jayfoad) [reply llvm/lib/Transforms/Utils/LowerSwitch.cpp:579]: I'm just trying to follow existing patterns. They all seem to use `assert` so won't do anything in a normal Release build. I don't think most compilers warn about unused _public_ fields do they?

**Comment by ruiling** [comment]
I think #68662 can help on this?

**Comment by jayfoad** [comment]
> I think #68662 can help on this?

If we go with #68662 then LowerSwitch is no longer used like an analysis, so there would be no need to verify that it is preserved.

**Comment by jayfoad** [comment]
> > I think #68662 can help on this?
> 
> If we go with #68662 then LowerSwitch is no longer used like an analysis, so there would be no need to verify that it is preserved.

We went with #68662.

**Pull request closed without merging**